components/toast_wrapper/toast_wrapper.tsx

```diff
@@ -2,103 +2,120 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
-import {FormattedMessage, injectIntl} from 'react-intl';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
 
 import Toast from 'components/toast/toast';
+
 import Timestamp, {RelativeRanges} from 'components/timestamp';
+
 import {isIdNotPost, getNewMessageIndex} from 'utils/post_utils';
+
 import * as Utils from 'utils/utils';
+
 import {isToday} from 'utils/datetime';
+
 import Constants from 'utils/constants';
+
 import {browserHistory} from 'utils/browser_history';
+
 import {SearchShortcut} from 'components/search_shortcut';
+
 import {HintToast} from 'components/hint-toast/hint_toast';
+
 import {Preferences} from 'mattermost-redux/constants';
+import {IDMappedObjects} from '@mattermost/types/utilities';
+import {Post} from '@mattermost/types/posts';
 
 const TOAST_TEXT_COLLAPSE_WIDTH = 500;
 
 const TOAST_REL_RANGES = [
     RelativeRanges.TODAY_YESTERDAY,
 ];
 
-class ToastWrapper extends React.PureComponent {
-    static propTypes = {
-        unreadCountInChannel: PropTypes.number,
-        newRecentMessagesCount: PropTypes.number,
-        channelMarkedAsUnread: PropTypes.bool,
-        isCollapsedThreadsEnabled: PropTypes.bool,
-        rootPosts: PropTypes.object,
-        atLatestPost: PropTypes.bool,
-        postListIds: PropTypes.array,
-        latestPostTimeStamp: PropTypes.number,
-        atBottom: PropTypes.bool,
-        lastViewedBottom: PropTypes.number,
-        width: PropTypes.number,
-        lastViewedAt: PropTypes.number,
-        focusedPostId: PropTypes.string,
-        initScrollOffsetFromBottom: PropTypes.number,
-        updateNewMessagesAtInChannel: PropTypes.func,
-        scrollToNewMessage: PropTypes.func,
-        scrollToLatestMessages: PropTypes.func,
-        scrollToUnreadMessages: PropTypes.func,
-        updateLastViewedBottomAt: PropTypes.func,
-        showSearchHintToast: PropTypes.bool,
-        onSearchHintDismiss: PropTypes.func,
-        shouldStartFromBottomWhenUnread: PropTypes.bool,
-        isNewMessageLineReached: PropTypes.bool,
-        unreadScrollPosition: PropTypes.string,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string,
-            }).isRequired,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * Action creator to update toast status
-             */
-            updateToastStatus: PropTypes.func.isRequired,
-        }).isRequired,
+type OwnProps = {
+    unreadCountInChannel: number;
+    newRecentMessagesCount: number;
+    channelMarkedAsUnread?: boolean;
+    isCollapsedThreadsEnabled?: boolean;
+    rootPosts?: IDMappedObjects<Post>;
```

Comment:
Checkout how we are using PropsFromRedux in `components/emoji_picker/emoji_picker.tsx`, that way we dont have to rewrite types from redux again.

Reply:
So if my understanding is correct: with PropsfromRedux we will be exporting a type ConnectedProps which will contain prop type definitions? In which case I will modify ownProps to extend this type.

Reply:
yes exactly, we dont have to type props arising from redux explicitly.